usl_models/usl_models/flood_ml/model.py

```diff
@@ -39,6 +34,7 @@ def __init__(
         self,
         model_params: FloodModelParams,
         spatial_dims: tuple[int, int] = (constants.MAP_HEIGHT, constants.MAP_WIDTH),
```

Comment:
If we're loading from a model check point, what does the user pass for model_params & spatial_dims? What happens if they're different from the params & dims used to train that model checkpoint?

I'm wondering if the FloodModel class is what we should be working with rather than working with the underlying keras model itself.

Reply:
The model params need to match what was used during export, otherwise it will throw an error referring to the shape of the weights.
We can't work with the loaded Keras model directly because we were unable to export its extra methods (call_n) and the signatures get mangled upon export (the dict becomes a kwargs mapping).

Reply:
so it sounds like we'd need to persist the `model_params` and `spatial_dims` as well to be able to reliably load the model? Or else get the call_n signature working -- do you see any obvious issues with what I'm doing wrong here:
https://github.com/UrbanSystemsLab/climateiq-cnn/pull/127
where I attempt to add that signature?

Reply:
Even if we had call_n serialization working, we still need to know the model_params and spatial_dims to provide correctly shaped input data. So the value of making this work for arbitrary model architecture is limitted to convenience of testing different hidden layers.
Testing changes to hidden layers is still easy enough when kicking off a custom training job, the only part that requires some thought is running test predictions, where the user must ensure that they are using the correct branch of the code so the model architecture lines up. 

We weren't able to get call_n to serialize without making major changes to the model code. I think we would need to get rid of the for loop and replace with tf.while_loop so the resulting tf.Graph doesn't unroll the loop for a fixed n during serialization. This will take some work and testing, so given the time constraint it's easiest to leave that out of serialization.

Reply:
okay, that sounds good -- let's just focus on documenting all the issues you've laid out here